### PR TITLE
feat: optional WITH KEY clause for SHOW TAG VALUES

### PR DESCRIPTION
**Pull request by lesam**

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package influxql
```
Comment (lesam) [inline utils.go:1]: I think this fits better here than in influxdb

**Comment by lesam** [comment]
For https://github.com/influxdata/influxdb/issues/15993

File: ast.go [hunk 2]
```diff
@@ -2849,6 +2849,12 @@ type ShowTagKeysStatement struct {
 	// Data sources that fields are extracted from.
 	Sources Sources
 
+	// Op to compare tag keys with
+	TagKeyOp Token
+
+	// Literal to compare tag keys with
+	TagKeyExpr Expr
+
```
Comment (dgnorton) [inline ast.go:2857]: Are `TagKeyOp` and `TagKeyExpr` specific to tags or do they also apply to field keys and expressions? I don't think InfluxQL can know during parsing whether it's a tag or field.

Comment (lesam) [reply ast.go:2857]: It gets re-written post-parsing as `_tagKey <TagKeyOp> <TagKeyExpr>` (or a string of OR expressions instead for WITH KEY IN).

**Review by dgnorton — APPROVED** [review]

**Pull request merged**